File: four_pillars_of_oop/no_01_encapsulation/no_02_maintainability/python/rectangle.py
class Rectangle:
    def __init__(self, width, height):
        self.__width = width  # Private attributes
        self.__height = height

    def set_width(self, width):
        if width > 0:
            self.__width = width
        else:
            print("Width must be positive.")

    def set_height(self, height):
        if height > 0:
            self.__height = height
        else:
            print("Height must be positive.")

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        return 2 * (self.__width + self.__height)

File: four_pillars_of_oop/no_01_encapsulation/no_02_maintainability/java/Rectangle.java
package four_pillars_of_oop.no_01_encapsulation.no_02_maintainability.java;

public class Rectangle {
    private double width;  // Private attributes
    private double height;

    public Rectangle(double width, double height) {
        if (width > 0 && height > 0) {
            this.width = width;
            this.height = height;
        }
    }

    public void setWidth(double width) {
        if (width > 0) {
            this.width = width;
        } else {
            System.out.println("Width must be positive.");
        }
    }

    public void setHeight(double height) {
        if (height > 0) {
            this.height = height;
        } else {
            System.out.println("Height must be positive.");
        }
    }

    public double getArea() {
        return width * height;
    }

    public double getPerimeter() {
        return 2 * (width + height);
    }
}
